extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -741,6 +762,34 @@ export class AmpA4A extends AMP.BaseElement {
   }
 
   /**
+   * Refreshes ad slot by fetching new creative and rendering it.
+   */
+  refresh() {
+    this.isRefreshing_ = true;
+    this.tearDownSlot();
```

Comment:
You need to handle multi-size and no-fill collapse cases:
tearDown will attempt to revert the slot size to its original which could fail if the slot is within the viewport.  If its collapsed it just means the ad request will never get sent (as there is code that detects if the slot has no height/width).  But if multi-size I assume it would fail because the sizes given could be greater than the slot?  We can handle these cases in one of two ways:
1.  Delay the refresh attempt until after the slot could be resized (would mean re-attempting resize once we know it could be successful when the slot is not in the viewport)
2.  For multi-size case, we could just remove sizes that are no longer eligible.